features/step_definitions/logging.rb

```diff
@@ -0,0 +1,128 @@
+  # first check packagemanifest exists for cluster-logging and elasticsearch-operator
+  required_packagemanifests = ['cluster-logging', 'elasticsearch-operator']
+  required_packagemanifests.each do |pm_name|
+    raise "Required packagemanifest #{pm_name} no found!" unless package_manifest(pm_name).exists?
+  end
+  # Create namespace
+  unless project('openshift-logging').exists?
+    # oc create -f https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/namespace.yaml
+    namespace_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/namespace.yaml"
+    @result = admin.cli_exec(:create, f: namespace_yaml)
+    raise "Error creating namespace" unless @result[:success]
+  end
+  logging_ns = "openshift-logging"
+  operators_ns = "openshift-operators"
+
+  # project('openshift-logging')
+  # Create operator group in `openshift-logging` namespace
+  operator_group_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/operator-group.yaml"
+  @result = admin.cli_exec(:create, f: operator_group_yaml, n: logging_ns)
+  raise "Error creating namespace" unless @result[:success]
+  # create catalogsourceconfig for logging and elasticsearch
+  catsrc_logging_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/csc-clusterlogigng.yaml"
+  catsrc_elasticsearch_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/csc-elasticsearch.yaml"
+
+  @result = admin.cli_exec(:create, f: catsrc_logging_yaml)
+  raise "Error creating catalogsourceconfig for cluster_logging" unless @result[:success]
+  @result = admin.cli_exec(:create, f: catsrc_elasticsearch_yaml)
+  raise "Error creating catalogsourceconfig for elasticsearch" unless @result[:success]
+
+  # create subscription in "openshift-operators" namespace:
+  sub_logging_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/sub-cluster-logging.yaml"
+  sub_elasticsearch_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/OCP-21311/sub-elasticsearch-operator.yaml"
+
+  @result = admin.cli_exec(:create, f: sub_logging_yaml) #, n: operators_ns)
+  raise "Error creating subscription for cluster_logging" unless @result[:success]
+  @result = admin.cli_exec(:create, f: sub_elasticsearch_yaml) #, n: operators_ns)
+  raise "Error creating subscription for elasticsearch" unless @result[:success]
+  # create instance in openshift-logging namespace
+  logging_yaml = "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/logging/clusterlogging/example.yaml"
+  @result = admin.cli_exec(:create, f: logging_yaml, n: logging_ns)
+  # Check the resource created by OLM
+  step %Q/I wait for clusterlogging to be functional in the project/
```

Comment:
shall we remove what we created at end of scenario? I see step is destructive tagged

Reply:
@akostadinov since with OLM install, the logging installation variations is much less than for ansible install for 3.x, I'm thinking for 4.x testing, we just install it once and leave it running.  @anpingli or @QiaolingTang should make a call on that front since they are the owner of those user stories.

Reply:
@pruan-rht  Most of time, we have to remove the clusterlogging/instance 1). it is better to clean it each time.
We needn't to clear the clusterlogging operator/elasticsearch operator and namespace each time.

1) To clean up clusterlogging, just remove the clusterlogging/instance resource  
#oc delete clusterlogging/instance -n openshift-logging
2) To clean up clusterlogging operator, we need to unsubscribe the operator
3) To clean up elasicsearch operator, we need to unsubscribe the operator
     
     
To clean up all resource

Reply:
@anpingli Thanks for the feedback.  If you do recommend removing after each install, perhaps we can add an option like `keep_installation: true` to the step and default to remove all previous installations to make it simpler.

Reply:
Looks good to have a option keep_installation: true

Reply:
Do you intend to make clean-up optional? Or do we drop the idea?

Reply:
@akostadinov @anli agrees with having that as optional.

Reply:
I missed that it is there, sorry.